libs/platform/src/lib/components/form/text-area/text-area.component.ts

```diff
@@ -0,0 +1,402 @@
+            throw new Error(`Textarea wrap type $ {this.wrapType} is not supported`);
+        }
+        // if not custom set, set counter to max length value, else it calculates remaining/exceeded characters.
+        if (!this.value) {
+            this.exceededCharCount = this.maxLength ? this.maxLength : 0;
+        } else {
+            this.isValueCustomSet = true;
+        }
+        super.ngOnInit();
+    }
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        // set max height
+        this._setMaxHeight();
+
+        // don't set any error state if we are not showing counter message
+        if (!this.showExceededText) {
+            this.stateType = null;
+        }
+
+        super.ngAfterViewInit();
+    }
+
+    /** @hidden */
+    ngAfterViewChecked() {
+        // when value is custom set(initial value is present), the heights don't get
+        // set to show the full text in the ngAfterViewInit immediately. therefore, we call autoGrowTextArea in
+        // ngAfterViewChecked and detect changes again.
+        if (this.value && this.isValueCustomSet) {
+            this.autoGrowTextArea(); // if applicable, grow the textarea when value is custom set
+            this.stateChanges.next('textarea: ngAfterViewChecked');
+        }
+    }
+
+    /** write value for ControlValueAccessor */
+    writeValue(value: any): void {
+        if (value) {
+            super.writeValue(value);
+            this.updateCounterInteractions();
+            this.stateChanges.next('textarea: writeValue');
+        }
+    }
+
+    /** update the counter message and related interactions */
+    updateCounterInteractions(): void {
+        if (this.value) {
+            this._textAreaCharCount = this.value.length;
+            if (this.maxLength) {
+                // newly added to avoid unnecessary iteration, remove if issue found
+                this.validateLengthOnCustomSet();
+            }
+        }
+    }
+
+    /** if exceeded maxlength when set as a value in code, highlight the exceeded text. */
+    validateLengthOnCustomSet(): void {
+        if (this._textAreaCharCount > this.maxLength) {
+            if (this._isPasted) {
+                this.textareaElement.nativeElement.focus();
+                // this.textareaElement.nativeElement.setSelectionRange(10, 20);
+                this.textareaElement.nativeElement.setSelectionRange(this.maxLength, this._textAreaCharCount);
+            }
+            this.counterExcessOrRemaining = this.excessText;
+            this.exceededCharCount = this._textAreaCharCount - this.maxLength;
+        } else {
+            this.counterExcessOrRemaining = this.remainingText;
+            this.exceededCharCount = this.maxLength - this._textAreaCharCount;
+        }
+        // reset flag
+        this._isPasted = false;
+    }
+
+    /** handle exceeded maxlength case when text is pasted */
+    handlePasteInteraction(): void {
```

Comment:
![textareacopypaste](https://user-images.githubusercontent.com/26483208/86121051-a3f89880-bad5-11ea-90c9-ac0868146f48.gif)
It's not working well.

Reply:
there is a requirement that if pasting results in exceeded characters, we need to highlight the exceeded text : 
> When the user pastes copied text, characters that are over the limit are selected automatically. The user can delete any excess characters by pressing Delete or Back on the keyboard (or virtual keyboard for phones and tablets).

In UI5 sample also we can see same behaviour: 
![2020-06-30_17-12-53 (1)](https://user-images.githubusercontent.com/53509521/86122370-5c820480-baf5-11ea-8f43-801d8922b688.gif)